src/components/auth/changePassword/ChangePasswordForm.tsx

```diff
@@ -22,20 +30,50 @@ const validationSchema: yup.SchemaOf<ChangePasswordFormData> = yup
       .min(6, customValidators.passwordMin)
       .required()
       .oneOf([yup.ref('password'), null], 'validation:password-match'),
+    token: yup.string(),
   })
 
 const defaults: ChangePasswordFormData = {
   password: '',
   confirmPassword: '',
+  token: '',
 }
 
 export type ChangePasswordFormProps = {
   initialValues?: ChangePasswordFormData
 }
 
 export default function ChangePasswordForm({ initialValues = defaults }: ChangePasswordFormProps) {
-  const onSubmit = (values: ChangePasswordFormData) => {
-    console.log(values)
+  const [loading, setLoading] = useState<boolean>(false)
+  const router = useRouter()
+  const token = router.query.token
+
+  const mutation = useMutation<AxiosResponse, AxiosError<ApiErrors>, ChangePasswordFormData>({
+    mutationFn: resetPassword(),
+    onError: (error) =>
+      AlertStore.show(
+        error.message == 'Request failed with status code 400'
+          ? 'Невалиден линк!'
+          : 'Потребителя не е открит, опитайте отново или се свържете със съпорта!',
```

Comment:
[defensive programming] no need to say what exactly happened to not help the attacker. Just say there was an error and try again or contact info@podkrepi.bg